File: com-mini-core/src/main/java/com/mini/security/digest/BaseDigest.java
package com.mini.security.digest;

import com.mini.security.SecurityBase;

import java.security.MessageDigest;

import static java.security.MessageDigest.getInstance;

/**
 * DigestUtil.java
 * @author xchao
 */
public abstract class BaseDigest extends SecurityBase {
    private final MessageDigest digest;

    @SuppressWarnings("WeakerAccess")
    protected BaseDigest(MessageDigest digest) {
        this.digest = digest;
    }

    @SuppressWarnings("WeakerAccess")
    protected BaseDigest(String algorithm) throws Exception {
        this(getInstance(algorithm));
    }

    public final BaseDigest update(byte[] input, int offset, int len) {
        digest.update(input, offset, len);
        return this;
    }

    public final BaseDigest update(byte[] input) {
        digest.update(input);
        return this;
    }

    public final BaseDigest update(byte input) {
        digest.update(input);
        return this;
    }

    public final byte[] digest(byte[] input) {
        return digest.digest(input);
    }

    public final byte[] digest() {
        return digest.digest();
    }

    public final String toString() {
        return digest.toString();
    }

    public final void reset() {
        digest.reset();
    }
}

File: mini-util/src/main/kotlin/com/mini/core/security/digest/BaseDigest.kt
@file:Suppress("unused")

package com.mini.core.security.digest

import com.mini.core.security.SecurityBase
import java.security.MessageDigest

/**
 * DigestUtil.java
 * @author xchao
 */
abstract class BaseDigest protected constructor(algorithm: String) : SecurityBase<BaseDigest>() {
    private val digest: MessageDigest = MessageDigest.getInstance(algorithm);

    override fun update(input: ByteArray, offset: Int, len: Int): BaseDigest {
        digest.update(input, offset, len)
        return this
    }

    override fun update(input: ByteArray): BaseDigest {
        digest.update(input)
        return this
    }

    override fun update(input: Byte): BaseDigest {
        digest.update(input)
        return this
    }

    override fun digest(input: ByteArray): ByteArray {
        return digest.digest(input)
    }

    override fun digest(): ByteArray {
        return digest.digest()
    }

    override fun toString(): String {
        return digest.toString()
    }

    fun reset() {
        digest.reset()
    }
}